### DISCUSSION TITLE
Linking to an issue without creating a reference

### DISCUSSION THREAD

**Discussion by bstinsonmhk** [community/community#23123] [New to GitHub]
<p>Is there a way to include a link in a commit message or PR that doesn’t create a reference on the target repository?</p>
<p>Say I’m developing a project at bstinsonmhk/foo and I rely on a library anotheruser/bar, Can I link to anotheruser/bar/issues/1234 without it showing up in their comment stream? They may not care that I’m referencing their issue for my own administrative purposes. </p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, there isn’t a way to prevent that cross-linking from occurring. On the flip side, that cross-link event is only visible to the people that have access to the repository from which the link originated. So if you link to a public repository from a private repository of your own, the only people that will be able to see the link event is yourself and whomever you grant access to your private repository.</p>
<p>I hope that helps!</p>

**Reply by James-E-A** [reply]
> No, there __isn’t a way__ to prevent that cross-linking from occurring. 

Per [comment #3239240](#discussioncomment-3239240) on June 18th, 2022, this answer is out of date.

**Reply by RaschidJFR** [reply]
It is now [officially supported](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls#avoiding-backlinks-to-linked-references) as mentioned in this [comment](https://github.com/orgs/community/discussions/23123#discussioncomment-11891227).

**Comment by JayFoxRox** [comment]
<p>Linking without references would be a useful feature.</p>
<p>I currently resort to just naming PRs by number without linking anymore (which is stupid).</p>
<p>I don’t want to cause a reference upstream when just prototyping something and I mention the related upstream issue.</p>
<p>On projects I maintain we also get stupid references like this in the discussion stream which just makes it harder to follow what’s important.</p>
<p>It has gotten to a point where I often ignore references, simply because &gt; 90% of them are accidental.</p>
<p>I think cross-repo links also shouldn’t produce a reference in discussion stream by default.</p>

**Comment by ichn-hu** [comment]
<p>I am recently bumping into this (literally today), would like to see this feature as a configuration.</p>

**Comment by samuelhwilliams** [comment]
<p>Ditto … I find this annoying/embarrassing when I just want to reference an issue in my PR and it creates some links in the issue itself. I’m just mentioning it for info, not for the upstream repo to do anything about.</p>

**Comment by rhcarvalho** [comment]
<p>I’ve seen usage of <a href="https://togithub.com/" rel="noopener nofollow ugc">https://togithub.com/</a> (a redirect domain) to achieve cross-linking without creating references.</p>
<p>This domain is what <a href="https://github.com/renovatebot/renovate" class="inline-onebox" rel="noopener nofollow ugc">GitHub - renovatebot/renovate: Universal dependency update tool that fits into your workflows.</a> uses when the bot makes comments.</p>
<p>Using an external domain for redirection is not a native GitHub feature tough, and of course the alternative domain may simply vanish or do evil without warning.</p>

**Comment by James-E-A** [comment]
<p>This is a horrible feature to not be able to opt-out of. I’m adding commits in my code to work around issues in someone else’s software; I want to properly document both what I’ve done and the paper trail behind why I’ve done it, but it seems <strong>seriously</strong> passive-aggressive to be piling events onto the issue I just reported… <em>“No. It’s fine. I’ll just polyfill it myself.”</em> :rofl:</p>

**Comment by andreineculau** [comment]
<p>FWIW one imperfect solution is to surround the link with backticks. This does in fact NOT create a link, but it’s not that annoying to copy and follow the link (some might even have mouse/keyboard shortcuts to do just that). I prefer that over relying that <a href="http://togithub.com" rel="noopener nofollow ugc">togithub.com</a> will have the same lifetime as <a href="http://github.com" rel="noopener nofollow ugc">github.com</a></p>

**Reply by tuurep** [reply]
Warning: Doesn't appear to work anymore, tried this today (21.10.2023) and it did create a reference

**Comment by sigprof** [comment]
<p>Is there still no official solution? Spamming issues and PRs with those “mentioned” references just makes me feel bad in some cases (and sometimes including the reference in a commit message results in further indirect spam when other people create copies of that commit by cherry-picking or rebasing).</p>
<p>I actually found a way to create a link to a PR that does not count as a mention without using a third-party domain — apparently replacing <code>github.com</code> with <code>www.github.com</code> in the URL still results in a working link (due to the automatic redirect to the non-<code>www</code> domain), but the “mentioned” reference does not appear for that link. E.g., I used <code>[Nix RFC 0101](https://www.github.com/NixOS/rfcs/pull/101)</code> to create a link to that PR from another PR in a public repository, and this did not result in a cross-link appearing in that PR.</p>
<p>One feature which gets broken by this change is the pop-up that usually appears when hovering over a link to a GitHub issue or PR — the link to <code>www.github.com</code> is treated as a plain hyperlink to some generic site, and the pop-up no longer appears.</p>
<p>Of course, GitHub owners could break this at any time by treating links to <code>www.github.com</code> exactly as <code>github.com</code> for the purpose of adding cross-links, but I hope that it would not happen. GitHub owners could also break the <code>www.github.com</code>→<code>github.com</code> redirect, but this style of redirection is really widespread, so there is some hope that it would stay.</p>

**Reply by danstewart** [reply]
This still works today. Thank you.

**Reply by yrro** [reply]
Just used this and I'm (re)confirming that it still works, cheers!

**Reply by ChillerDragon** [reply]
Nice hack thanks. Still works in 2024. Be careful putting just hashtag and the number (not the full url) like ``#101`` in the issue/pr title will not create a link. But as comment or commit it will create a link. So if you post a gdb backtrace in your commit message or try to link your downstream issue in a comment it will create links in the upstream. You can see one example here https://github.com/NixOS/nixpkgs/pull/1

**Reply by ghost** [reply]
Here is a very simple git hook, that will prevent backlinks from commit messages in the repo you place it in.
You will have to do this per machine and per repo though (although using a git template can help with that).
https://gist.github.com/sadamczyk/813e15fba295b17a06fe946d548f7491

It just replaces every `github.com` link with `www.github.com`. 
But it's good enough for some peace of mind that I'm not unnecessarily spamming random PRs / issues with backlinks from my personal dotfiles repo, especially when rebasing repeatedly...

**Reply by rafaeljcd** [reply]
many thanks!

**Reply by osa1** [reply]
This still works to avoid creating a backlink, but you don't get the same type of link, with a green/purple dot indicating whether the issue is still open or not.

**Reply by sid-maddy** [reply]
Using a relative link (like `[this](/grafana/alloy/issues/474)`) seems to work, too.

> [!NOTE]
> Obviously, this only works, if the link is on GitHub.

**Comment by adamerose** [comment]
So just to clarify the problem... When a comment in issue A references issue B:
- Issue B will display a message like `ExampleUser mentioned this issue X minutes ago` with a link to issue A
- Editing or deleting the comment does not remove the message
- Completely deleting the issue _does_ remove the message
- The only other way to get this message removed is getting help from GitHub support admins
- There is no way for the owner of the repository with issue B to delete these messages

I would hope to see GitHub make these changes:
 - Add the ability to delete messages about comments referencing issues on repositories you own
 - Make this referencing behavior opt-in instead of automatic
 - Update the messages when the originating comment gets edited or deleted

**Comment by polarathene** [comment]
Probably the only way to get Github to address it is when it poses more of a problem to raise priority? :confused: (_malicious spam activity to the point that banning accounts is ineffective?_ :man_shrugging: )

---

I've seen various lengthy PRs in the past get affected by this issue unintentionally. Sometimes the commit author may not even be aware everytime they rewrite history that updates the affected commit hash to a new one it appends another reference to the issue or PR, CI could make that a frequent event to carry a patch for example.

While that can pollute an issue or PR discussion (_mostly problematic when it's enough times that it triggers the web UI to hide earlier items like actual comments that have much more value_), I've also witnessed a related problem with `@username` mentions.

Sometimes I am mentioned in a commit message and this will trigger a notification. The repo can be cloned and published as a new repo instead of forked, and this will likewise apply any commit message references including those mention notifications. If there is a CI involved that's rewriting commit history that can be quite annoying :sweat_smile: (_again usually unintentional, but I've also seen attempts to steal and rebrand projects by rewriting commit authors where the ping was useful_)

**Reply by iTrooz** [reply]
It would be great if in an issue, every reference to a commit (defined by its code changes, or metadata other than the hash) were stacked into a single "message", collapsed by default. And further deduplicate links in this collapsed message.

**Comment by yezhengli-Mr9** [comment]
permalink not work in issues of my personal github repo?
https://github.com/orgs/community/discussions/54598

**Comment by mschoettle** [comment]
You can use `redirect.github.com` instead of `github.com` in your reference URL. This is what dependabot and Renovate Bot use in release notes. I am trying to get this documented so it is easier to find: https://github.com/github/docs/pull/35987
